rust/rope/src/tree.rs

```diff
@@ -179,6 +179,14 @@ impl<N: NodeInfo> Node<N> {
     fn height(&self) -> usize {
         self.0.height
     }
+    
+    fn is_leaf(&self) -> bool {
+        if let NodeVal::Leaf(_) = self.0.val {
```

Comment:
Oh, I was about to do that, and even wanted to debug_assert that the height is 0 when the val is a Leaf. I'll just change it to height == 0, because it's so short :)